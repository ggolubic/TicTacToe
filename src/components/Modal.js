import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

import Score from "./Score";
import { Title } from "../App";
import Button from "./Button";

const Modal = ({ score, newGame, resetScore, winner }) => {
  return (
    <ModalContainer>
      <ModalContent>
        <Title>{winner ? `You won!` : `Its a draw!`}</Title>
        <div>
          <ScoreArea>
            <SubTitle>Current Score:</SubTitle>
            <Score score={score} />
          </ScoreArea>

          <ModalNewGame>
            <Button new onClick={() => newGame()}>
              New Round
            </Button>
            <Button onClick={() => resetScore()}>Reset Score</Button>
          </ModalNewGame>
        </div>
      </ModalContent>
    </ModalContainer>
  );
};

Modal.defaultProps = {
  score: { x: 0, o: 0, draw: 0 },
  winner: null,
  newGame: () => {},
  resetScore: () => {}
};

Modal.propTypes = {
  score: PropTypes.shape({
    x: PropTypes.number,
    o: PropTypes.number,
    draw: PropTypes.number
  }).isRequired,
  winner: PropTypes.oneOf([null, "x", "o"]),
  newGame: PropTypes.func.isRequired,
  resetScore: PropTypes.func.isRequired
};

export default Modal;
const ModalContainer = styled.div`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0%;
  z-index: 1;
  background: rgba(0, 0, 0, 0.5);
`;

const ModalContent = styled.div`
  position: absolute;

  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  border-radius: 5%;
  width: 450px;
  height: 400px;
  background: white;
  transition: all 350ms cubic-bezier(0.68, -0.55, 0.265, 1.18);
`;

const ScoreArea = styled.div`
  text-align: center;
`;

const SubTitle = styled.h2`
  font-family: "Montserrat", sans-serif;
  text-align: center;
`;

const ModalNewGame = styled.div`
  text-align: center;
  margin-top: 50px;
`;
