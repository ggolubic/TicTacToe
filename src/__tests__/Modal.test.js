import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import "jest-styled-components";

import Modal from "../components/Modal";

Enzyme.configure({ adapter: new Adapter() });

describe("Modal component", () => {
  it("Should render correctly with default props", () => {
    const tree = renderer.create(<Modal />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should render correctly with passed props", () => {
    const props = setup();
    const tree = renderer.create(<Modal {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Should call newGame when first button is clicked", () => {
    const { newGame, ...props } = setup();
    const component = mount(<Modal newGame={newGame} {...props} />);

    // Test before event
    expect(newGame).not.toHaveBeenCalled();

    // simulate the click event
    component
      .find("button")
      .first()
      .props()
      .onClick();
    // Test after event
    expect(newGame).toHaveBeenCalled();

    component.unmount();
  });
  it("Should call resetScore when second button is clicked", () => {
    const { resetScore, ...props } = setup();
    const component = mount(<Modal resetScore={resetScore} {...props} />);

    // Test before event
    expect(resetScore).not.toHaveBeenCalled();

    // simulate the click event
    component
      .find("button")
      .last()
      .props()
      .onClick();
    // Test after event
    expect(resetScore).toHaveBeenCalled();

    component.unmount();
  });
});

function setup() {
  const props = {
    score: { x: 0, o: 0, draw: 0 },
    winner: null,
    newGame: jest.fn(),
    resetScore: jest.fn()
  };
  return { ...props };
}
