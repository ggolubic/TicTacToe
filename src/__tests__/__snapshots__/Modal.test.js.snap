// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component Should render correctly with default props 1`] = `
.c6 {
  color: #545454;
}

.c7 {
  color: #ffd02d;
}

.c5 {
  font-family: "Montserrat",sans-serif;
  width: 300px;
  margin: 0 auto;
  text-align: center;
  margin-top: 50px;
}

.c9 {
  font-family: "Montserrat",sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  outline: none;
  background: #ddbc95;
  color: white;
}

.c9:hover {
  color: white;
  background: #ddbc95;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c9:hover {
  color: #ddbc95;
  background: white;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c10 {
  font-family: "Montserrat",sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  outline: none;
}

.c10:hover {
  color: white;
  background: #ddbc95;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c2 {
  font-family: "Montserrat",sans-serif;
  text-align: center;
}

.c0 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0%;
  z-index: 1;
  background: rgba(0,0,0,0.5);
}

.c1 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 5%;
  width: 450px;
  height: 400px;
  background: white;
  -webkit-transition: all 350ms cubic-bezier(0.68,-0.55,0.265,1.18);
  transition: all 350ms cubic-bezier(0.68,-0.55,0.265,1.18);
}

.c3 {
  text-align: center;
}

.c4 {
  font-family: "Montserrat",sans-serif;
  text-align: center;
}

.c8 {
  text-align: center;
  margin-top: 50px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      Its a draw!
    </h1>
    <div>
      <div
        className="c3"
      >
        <h2
          className="c4"
        >
          Current Score:
        </h2>
        <div
          className="c5"
        >
          <h2
            className="c6"
          >
            Player 
            1
            : 
            0
          </h2>
          <h2
            className="c7"
          >
            Player 
            2
            : 
            0
          </h2>
          <h2>
            Draw: 
            0
          </h2>
        </div>
      </div>
      <div
        className="c8"
      >
        <button
          className="c9"
          onClick={[Function]}
        >
          New Round
        </button>
        <button
          className="c10"
          onClick={[Function]}
        >
          Reset Score
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal component Should render correctly with passed props 1`] = `
.c6 {
  color: #545454;
}

.c7 {
  color: #ffd02d;
}

.c5 {
  font-family: "Montserrat",sans-serif;
  width: 300px;
  margin: 0 auto;
  text-align: center;
  margin-top: 50px;
}

.c9 {
  font-family: "Montserrat",sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  outline: none;
  background: #ddbc95;
  color: white;
}

.c9:hover {
  color: white;
  background: #ddbc95;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c9:hover {
  color: #ddbc95;
  background: white;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c10 {
  font-family: "Montserrat",sans-serif;
  border-radius: 4%;
  background: transparent;
  border: 2px solid #ddbc95;
  color: #ddbc95;
  margin: 0 1rem;
  font-size: 1rem;
  font-weight: 700;
  padding: 0.5rem 1rem;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
  outline: none;
}

.c10:hover {
  color: white;
  background: #ddbc95;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c2 {
  font-family: "Montserrat",sans-serif;
  text-align: center;
}

.c0 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0%;
  z-index: 1;
  background: rgba(0,0,0,0.5);
}

.c1 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 5%;
  width: 450px;
  height: 400px;
  background: white;
  -webkit-transition: all 350ms cubic-bezier(0.68,-0.55,0.265,1.18);
  transition: all 350ms cubic-bezier(0.68,-0.55,0.265,1.18);
}

.c3 {
  text-align: center;
}

.c4 {
  font-family: "Montserrat",sans-serif;
  text-align: center;
}

.c8 {
  text-align: center;
  margin-top: 50px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      Its a draw!
    </h1>
    <div>
      <div
        className="c3"
      >
        <h2
          className="c4"
        >
          Current Score:
        </h2>
        <div
          className="c5"
        >
          <h2
            className="c6"
          >
            Player 
            1
            : 
            0
          </h2>
          <h2
            className="c7"
          >
            Player 
            2
            : 
            0
          </h2>
          <h2>
            Draw: 
            0
          </h2>
        </div>
      </div>
      <div
        className="c8"
      >
        <button
          className="c9"
          onClick={[Function]}
        >
          New Round
        </button>
        <button
          className="c10"
          onClick={[Function]}
        >
          Reset Score
        </button>
      </div>
    </div>
  </div>
</div>
`;
